PORTAL/jobs/_common.py

```diff
@@ -160,7 +170,8 @@ class JobFS(types.SimpleNamespace):
     This serves as the base class for context-specific subclasses.
     """
 
-    context: Optional[str] = None  # required in subclasses
+    CONTEXT: Optional[str] = None  # required in subclasses
```

Comment:
```suggestion
    CONTEXT: str = ''  # required in subclasses
```

This should make the type checker happy -- otherwise we can revert all of the `context: str` to `context: Optional[str]`.